### PR TITLE
build: include header "unit.h" after c++ headers

### PR DESCRIPTION
**Pull request by drewdzzz**
Header "unit.h" contains `ok` and `is` macros used to check test cases. The problem is such simple names can be used in C++ STL library headers (it's OK because such short names can be hidden in a namespace), so when including, for example, header "vector" after "unit.h", build can fail because function declaration or definition in the C++ header will turn into a macro invocation. I faced this problem building Tarantool on MacOS with SDK of 14.4 version.

### CODE REVIEW

**Review by sergepetrenko — APPROVED** [review]
LGTM. Please, reassign me once you get an approval from devX

**Review by ochaplashkin — APPROVED** [review]

**Comment by sergepetrenko** [comment]
@drewdzzz, please also add the 'full-ci' lable after assigning the PR on me for merging.

**Comment by sergepetrenko** [comment]
Backports:
2.11 https://github.com/tarantool/tarantool/pull/9886
3.0 https://github.com/tarantool/tarantool/pull/9887

**Pull request merged**
